src/Appwrite/Platform/Modules/Functions/Workers/Builds.php

```diff
@@ -445,7 +445,7 @@ protected function buildDeployment(
                     Console::execute('rsync -av --exclude \'.git\' ' . \escapeshellarg($tmpTemplateDirectory . '/' . $templateRootDirectory . '/') . ' ' . \escapeshellarg($tmpDirectory . '/' . $rootDirectory), '', $stdout, $stderr);
 
                     // Commit and push
-                    $exit = Console::execute('git config --global user.email "team@appwrite.io" && git config --global user.name "Appwrite" && cd ' . \escapeshellarg($tmpDirectory) . ' && git add . && git commit -m "Create ' . \escapeshellarg($resource->getAttribute('name', '')) . ' function" && git push origin ' . \escapeshellarg($branchName), '', $stdout, $stderr);
+                    $exit = Console::execute('git config --global user.email "team@appwrite.io" && git config --global user.name "Appwrite" && cd ' . \escapeshellarg($tmpDirectory) . ' && git checkout -b ' . \escapeshellarg($branchName) . ' && git add . && git commit -m "Create ' . \escapeshellarg($resource->getAttribute('name', '')) . ' function" && git push origin ' . \escapeshellarg($branchName), '', $stdout, $stderr);
```

Comment:
won't this error out if the branch already exists?

Reply:
I wanted to prove the problem is there, but couldnt. we might be using speciifc git cli version that simply makes it work.

![CleanShot 2025-07-01 at 11 10 17@2x](https://github.com/user-attachments/assets/ede3e129-85a7-42fd-8555-01411a44c54f)

![CleanShot 2025-07-01 at 11 12 02@2x](https://github.com/user-attachments/assets/51f64365-8694-4554-8dd2-f676838186c1)